saltproc/app.py

```diff
@@ -0,0 +1,464 @@
+from saltproc import Depcode
+from saltproc import Simulation
+from saltproc import Materialflow
+from saltproc import Process
+from saltproc import Reactor
+# from depcode import Depcode
+# from simulation import Simulation
+# from materialflow import Materialflow
+import os
+import copy
+import json
+from collections import OrderedDict
+import gc
+import networkx as nx
+import pydotplus
+import argparse
+import numpy as np
+
+
+input_path = os.path.dirname(os.path.abspath(__file__))
+
+input_file = os.path.join(input_path, 'data/saltproc_tap')
+iter_matfile = os.path.join(input_path, 'data/saltproc_mat')
+
+
+def check_restart(restart_flag):
+    """If the user set `Restart simulation from the step when it stopped?`
+    for `False` clean out iteration files and database from previous run.
+
+    Parameters
+    ----------
+    restart_flag : bool
+        Is the current simulation restarted?
+    """
+    if not restart_flag:
+        try:
+            os.remove(db_file)
+            os.remove(iter_matfile)
+            os.remove(input_file)
+        except OSError as e:
+            print("Error while deleting file: ", e)
+
+
+def parse_arguments():
+    """Parses arguments from command line.
+
+    Parameters
+    -----------
+
+    Returns
+    --------
+    n: int
+        Number of nodes for use in Serpent simulation.
+    d: int
+        Number of cores for use in Serpent simulation.
+    i: str
+        Path and name of main SaltProc input file (json format).
+
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-n',      # Number of nodes to use
+                        type=int,
+                        default=1,
+                        help='number of cluster nodes to use in Serpent')
+    parser.add_argument('-d',      # Number of nodes to use
+                        type=int,
+                        default=1,
+                        help='number of threads to use in Serpent')
+    parser.add_argument('-i',      # main input file
```

Comment:
It might be useful to add the functionality where the user can define the name of the output file: `-o output.txt` or something